src/cmd-build

```diff
@@ -231,7 +231,13 @@ composejson=${PWD}/tmp/compose.json
 # Put this under tmprepo so it gets automatically chown'ed if needed
 lockfile_out=${tmprepo}/tmp/manifest-lock.generated.${basearch}.json
 prepare_compose_overlays
-# --cache-only is here since `fetch` is a separate verb.
+# --cache-only is here since `fetch` is a separate verb; however, we do a fetch
+# explicitly if there appears to be no cache, in order to allow people to just
+# do `cosa build` on a clean dir.
+if [ ! -f "${workdir}"/builds/builds.json ] && [ ! -f "${workdir}/cache/fetched-stamp" ] ; then
+    echo "No previous build or fetch detected; automatically fetching"
+    (cd "${workdir}" && cosa fetch)
```

Comment:
we may or may not want `--strict` right here (#1378)

Reply:
Yeah, we should pass `--strict` here based on whether `cosa build` itself was passed `--strict`.